File: sample/src/main/java/cc/coopersoft/android/keycloak/KeycloakClient.java
package cc.coopersoft.android.keycloak;

import cc.coopersoft.accesstoken.keycloak.KeycloakTokenHolder;
import cc.coopersoft.accesstoken.keycloak.KeycloakTokenStorage;


public class KeycloakClient {

  private volatile static KeycloakTokenHolder accessTokenHolder;

  public static void init(KeycloakTokenStorage storage){
    accessTokenHolder = new KeycloakTokenHolder.Builder()
            .host("www.XXX.com") // root address: like "www.XXX.com"
            .realms("realms")  // realms: config in your keycloak
            .clientId("client") // clientId: config in your keycloak
            .clientSecret("XXXXX-XXXXX-XXXX-XXXX-1XXXXX") //clientSecret : clientId: config in your keycloak
            .storage(storage) // storage: you can custom for cc.coopersoft.accesstoken.keycloak.KeycloakTokenStorage
            .build();
  }

  public static KeycloakTokenHolder accessTokenHolder() {
    if (accessTokenHolder == null) {
      synchronized (KeycloakClient.class) {
      }
    }
    if (accessTokenHolder == null) {
      throw new IllegalStateException("must init first");
    }
    return accessTokenHolder;
  }
}

File: sample/src/main/java/cc/coopersoft/android/keycloak/KeycloakClient.kt
package cc.coopersoft.android.keycloak

import cc.coopersoft.accesstoken.keycloak.KeycloakTokenHolder
import cc.coopersoft.accesstoken.keycloak.KeycloakTokenStorage

object KeycloakClient {
    @Volatile
    private var accessTokenHolder: KeycloakTokenHolder? = null
    @JvmStatic
    fun init(storage: KeycloakTokenStorage?) {
        accessTokenHolder = KeycloakTokenHolder.Builder()
            .host("www.XXX.com")
            .preRealmsPath(null)
            .realms("realms")
            .clientId("client")
            .clientSecret("XXXXX-XXXXX-XXXX-XXXX-1XXXXX")
            .storage(storage)
            .build()
    }

    @JvmStatic
    fun accessTokenHolder(): KeycloakTokenHolder? {
        if (accessTokenHolder == null) {
            synchronized(KeycloakClient::class.java) {}
        }
        checkNotNull(accessTokenHolder) { "must init first" }
        return accessTokenHolder
    }
}